### PR TITLE
Cleanup docstrings for `gammapy.catalog`

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This addresses part of issue https://github.com/gammapy/gammapy/issues/4811, specifically for `gammapy.catalog`

### CODE REVIEW

**Review by QRemy — DISMISSED** [review]
Thanks a lot @Astro-Kirsty, no comment from my side.

**Review by registerrier — DISMISSED** [review]
Thanks @Astro-Kirsty . This looks good.

Have you tried running pydocstyle to check?

**Comment by Astro-Kirsty** [comment]
I ran pydocstyle, and there are few warnings. Specifically when we do not specify a docstring, we actually tend to have a number of functions without docstring (lots of `__init__` functions). Another common warning is when we have a one line docstring that references a 'signature'.

pydocstyle warning: `D402: First line should not be the function's "signature"`
Examples related to error: 
`"""Source positions (`~astropy.coordinates.SkyCoord`)."""`
`"""Catalog description (str)."""`

The explanation of fixing this can be found here
https://peps.python.org/pep-0257/#one-line-docstrings

File: gammapy/catalog/core.py [hunk 1]
```diff
@@ -274,7 +274,7 @@ def _lookup_extended_source_idx(self):
 
     @property
     def positions(self):
-        """Source positions (`~astropy.coordinates.SkyCoord`)."""
+        """Source positions as a list of `~astropy.coordinates.SkyCoord` objects."""
```
Comment (QRemy) [inline gammapy/catalog/core.py:277]:
```suggestion
        """Source positions a `~astropy.coordinates.SkyCoord` object."""
```
it's not a list of SkyCoord objects but only one that includes a list of coordinates

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty !

**Pull request merged**